src/_pytest/assertion/rewrite.py

```diff
@@ -854,6 +873,39 @@ def visit_Assert(self, assert_):
             set_location(stmt, assert_.lineno, assert_.col_offset)
         return self.statements
 
+    def warn_about_none_ast(self, node, module_path, lineno):
+        """Returns an ast warning if node is None with the following statement:
+        if node is None:
+            from _pytest.warning_types import PytestWarning
+            import warnings
+            warnings.warn_explicit(
+                PytestWarning('assertion the value None, Please use "assert is None"'),
+                category=None,
+                # filename=str(self.module_path),
+                filename=__file__
+                lineno=node.lineno,
+            )
+        """
+
+        # using parse because it's different between py2 py3
+        AST_NONE = ast.parse("None").body[0].value
+        val_is_none = ast.Compare(node, [ast.Is()], [AST_NONE])
+        send_warning = ast.parse(
+            """
+from _pytest.warning_types import PytestWarning
+from warnings import warn_explicit
+warn_explicit(
+    PytestWarning('assertion the value None, Please use "assert is None"'),
+    category=None,
+    filename='{filename}',
```

Comment:
I get this error on Windows:

```
  File "c:\users\bruno\pytest\src\_pytest\assertion\rewrite.py", line 843, in visit_Assert
    top_condition, module_path=self.module_path, lineno=assert_.lineno
  File "c:\users\bruno\pytest\src\_pytest\assertion\rewrite.py", line 904, in warn_about_none_ast
    filename=str(module_path), lineno=lineno
  File "C:\Users\bruno\AppData\Local\Programs\Python\Python36-32\lib\ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 7
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \uXXXX escape
```

The problem is using `str(module_path)` here, because this will generate a string with `\` characters and potential invalid escapes.

We should use this instead:

```
warn_explicit(
    PytestWarning('assertion the value None, please use "assert is None"'),
    category=None,
    filename={filename!r},
    lineno={lineno},
)
            """.format(
                filename=module_path, lineno=lineno
```

Reply:
_Ooh_ I was wondering how to fix that! Python string formatting is truly a pit with no end

Reply:
Heh!

Also, I just realized that you should use `filename=module_path.strpath`, otherwise we will get the `py.path.local` representation in the code, which is not what we want.